File: platform/packages_mjs/env-resolver/src/env-confluence.x.ts
import { resolve } from '@internal/env-resolve';

export interface ConfluenceEnv {
    baseUrl: string | undefined;
    username: string | undefined;
    apiToken: string | undefined;
}

export function resolveConfluenceEnv(config?: Record<string, any>): ConfluenceEnv {
    return {
        baseUrl: resolve(undefined, ['CONFLUENCE_BASE_URL'], config, 'baseUrl', undefined),
        username: resolve(undefined, ['CONFLUENCE_USERNAME'], config, 'username', undefined),
        apiToken: resolve(undefined, ['CONFLUENCE_API_TOKEN'], config, 'apiToken', undefined),
    };
}

File: platform/packages_py/env_resolver/env_resolver/env_confluence_x.py
from dataclasses import dataclass
from typing import Any

from env_resolve import resolve


@dataclass(frozen=True)
class ConfluenceEnv:
    base_url: str | None
    username: str | None
    api_token: str | None


def resolve_confluence_env(config: dict[str, Any] | None = None) -> ConfluenceEnv:
    return ConfluenceEnv(
        base_url=resolve(None, ['CONFLUENCE_BASE_URL'], config, 'base_url', None),
        username=resolve(None, ['CONFLUENCE_USERNAME'], config, 'username', None),
        api_token=resolve(None, ['CONFLUENCE_API_TOKEN'], config, 'api_token', None),
    )
